### ISSUE TITLE
Colour assignation

### ISSUE THREAD

**Issue by SachaDef** [SnaKyEyeS/ADE-Scheduler#319]
**Is your feature request related to a problem? Please describe.**
I'm bulding my optimal schedule using groups of classes rather than individual classes, such as 'FSA12BA', 'FILIERE INFO 12' and 'FILIERE ELEC 12' in my case. Bulding my schedule for the 2nd quarter, I have now unchecked all the classes from the 1st quarter of 'FSA12BA'. However, they are still assigned a colour in the colour palette, and because there are too much classes in my schedule, there is a colour overflow and two of the classes I left checked now have the same colour.
![img3](https://user-images.githubusercontent.com/65407167/105344019-aee65800-5be2-11eb-96ab-90e18cc0d5b5.png)

**Describe the solution you'd like**
Be able to assign a colour from the palette to a specific class. I do not mean change the colours available in the palette, as you can do in the 'Account' section, but rather be able to tell which class would get assigned to which available colour in the palette.

**Describe alternatives you've considered**
Unassign colours to totally unchecked classes, so that they leave some colours free for the actual checked and desired classes. This seemed more complicated though.

**Additional context**
This feature is not particularly important or crucial, as all I have to do to fix it is select individual classes rather than groups of classes, as to not overflow the palette.
Thanks a lot for your work !

**Comment by jeertmans** [comment]
Recently, I received many ideas to improve the current system of Colors.

From that, this is what I propose to do : 
1 - Add possibility to add colors to the colors palette
2 - Create 3 differents color mode with a toggle to switch between them : 
    2.1 - The current mode (one color per course in the order they appear)
    2.2 - One color per event type (as asked by a friend of mine)
    2.3 - One color but assigned per course (so the colors palette disappears and you can change the color of the course by clicking on their summary)

What do you think @SachaDef @louisna @SnaKyEyeS ?

**Comment by SachaDef** [comment]
I think it would be very polyvalent (versatile ?), and would be able to satisfy everyone's little preferences with even more customization !

**Comment by SnaKyEyeS** [comment]
I disagree with the idea to implement a toggle to switch between various modes (it is too much for too few IMO).

I’d rather settle on one good color mode: one color per course, but I am not 100% certain how we would implement this while ensuring backward compatibility & not-to-messy code....

**Comment by jeertmans** [comment]
@SnaKyEyeS that is not because you will not use it that other people won't :)
I know it may be hard to implement but that maybe the most recurring request I get when I ask "What could we improve".

Modes 1 and 2 are easy, more 3 is challenging but not impossible.
